src/Algebra/Graph/AdjacencyMap.hs

```diff
@@ -754,6 +755,20 @@ gmap f = AM . Map.map (Set.map f) . Map.mapKeysWith Set.union f . adjacencyMap
 induce :: (a -> Bool) -> AdjacencyMap a -> AdjacencyMap a
 induce p = AM . Map.map (Set.filter p) . Map.filterWithKey (\k _ -> p k) . adjacencyMap
 
+-- | Construct the /induced subgraph/ of a given graph by removing the 
+-- vertices that are Nothing.
+-- Complexity: /O(n * log(n))/ time.
+-- @
+-- induceJust ('vertex' (Nothing :: Maybe Int))             == 'empty'
+-- induceJust (gmap Just x)                                 == x
+-- induceJust ('connect' (gmap Just x) ('vertex' Nothing))  == x
+-- @
+induceJust :: Ord a => AdjacencyMap (Maybe a) -> AdjacencyMap a
+induceJust = AM . Map.map catMaybesSet . catMaybesMap . adjacencyMap
+    where 
+      catMaybesSet = Set.map Maybe.fromJust . Set.delete Nothing
+      catMaybesMap = Map.mapKeys Maybe.fromJust . Map.delete Nothing
```

Comment:
Can we avoid using partial functions like `fromJust`? Perhaps, you could use `Map.traverseMaybeWithKey`? Or would it be much slower?

(Also in other modules.)

Reply:
I just tried and unfortunately we can't do it because it doesn't provide a way to change `key`: 
```haskell
traverseMaybeWithKey :: Applicative f => (k -> a -> f (Maybe b)) -> Map k a -> f (Map k b)
```
It's too bad because this function is `O(n)` instead of `O(n * log(n))` like `mapKeys`

`Map.fromJust` raises red flags but it should be safe because there can be only one `Nothing` key, right?

Reply:
I think we could use `Set.mapMonotonic` and `Map.mapKeysMonotonic` to get O(n) complexity.

Reply:
This will rely on the existing `Ord` instance for `Maybe`, which preserves the order in `Just`s.

Reply:
Awesome, seems like it works. :)